### ISSUE TITLE
logstr initialization should happen before calling makeConnection in tcp.BaseClient

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2813]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#2813|
|Type|enhancement|
|Created|2007-09-05 13:11:02Z|

That would prevent the ```[Uninitialized]``` when you call log in connectionMade.

<details><summary>Searchable metadata</summary>

```
trac-id__2813 2813
type__enhancement enhancement
reporter__therve therve
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1188997862000000 1188997862000000
changetime__1193994332000000 1193994332000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [21284]) Simple fix, more complex test.

Refs [#2813](https://github.com/twisted/twisted/issues/2813)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [21285]) Whitespaces and imports cleanups.

Refs [#2813](https://github.com/twisted/twisted/issues/2813)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve removed owner|
|-|-|

Ready to review in logstr-init-2813.

**Comment by twisted-trac** [comment]
|[<img alt="Peaker's avatar" src="https://avatars.githubusercontent.com/u/48899?s=50" width="50" height="50">](https://github.com/Peaker)<a name="note_4"></a>|@Peaker commented|
|-|-|

A superficial look seems good to me.

However, I think the test could be more meaningful if it did verify equality when it still equals, before verifying inequality (as a control verification).

I don't know enough about the meaning of logPrefix and other things in there to give an "OK" though.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

(In [21353]) Check equality using the connector.

Refs [#2813](https://github.com/twisted/twisted/issues/2813)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

Merged forward to **logstr-init-2813-2** because of disgusting buildbot/trial integration failure.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @therve|
|-|-|

It'd be great to have more specific assertions about the value of logstr.  Testing against "Uninitialized" and "MyProtocol,client" explicitly makes sense to me.

As long as you're here, it might not hurt to add a test for tcp.Server's logstr attribute? :)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve commented|
|-|-|

(In [21516]) Process review comments.

Refs [#2813](https://github.com/twisted/twisted/issues/2813)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve removed owner|
|-|-|

Thanks, back to review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @therve|
|-|-|

The new test fails on iocp.  Doesn't particularly bother me.  Fix if you like, otherwise you can merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_11"></a>|@therve set status to `closed`|
|-|-|

(In [21531]) Merge logstr-init-2813-2

Author: therve
Reviewer: exarkun
Fixes [#2813](https://github.com/twisted/twisted/issues/2813)

Make the initialization of the log prefixed used by tcp.BaseClient
happen earlier, so that the value is set when the connectionMade is called
on the client protocol.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
